lib/recurly/hosted-fields.js

```diff
@@ -63,7 +67,17 @@ export class HostedFields extends Emitter {
   }
 
   inject () {
-    this.on('hostedField:ready', this.readyHandler.bind(this));
+    this.on('hostedField:ready', this.onReady.bind(this));
+
+    if (this.serveAsSingleField()) {
+      const fieldType = 'card';
+      const fieldConfig = this.fieldConfig(fieldType);
+      Object.assign(fieldConfig, { singleField: true });
+      this.fields.push(new HostedField(fieldConfig));
+      this.initQueue.push(fieldType);
+      return;
+    }
```

Comment:
There are far too many assumptions here about field type and I don't like that the fieldConfig is being mutated here. It should really be static from the time it is read from `this.fieldConfig`, else that method is incomplete. This routine is also somewhat redundant given the iterator below.

Reply:
I think it's ultimately just not necessary to detect whether there are one or many hosted fields, since they should all get just one tabbing sink :)